lib/app.ts

```diff
@@ -1,16 +1,19 @@
+import { Channel } from './channel'
+import { AppConfigAPI, AppState } from './types'
+
 const HOOK_STAGE_PRE_INSTALL = 'preInstall'
 const HOOK_STAGE_POST_INSTALL = 'postInstall'
 
-const isObject = o => typeof o === 'object' && o !== null && !Array.isArray(o)
-const isFunction = f => typeof f === 'function'
-const isPromise = p => isObject(p) && isFunction(p.then)
+const isObject = (o: any) => typeof o === 'object' && o !== null && !Array.isArray(o)
```

Comment:
I wouldn't move it to a future PR, because the ticket likely wouldn't be worth the overhead of checking all the types again. I'll try using type guards.
On that note: I think there's a few tests that check for correct types on return etc. that aren't doing much since we're using typescript. Should I remove those?